### DISCUSSION TITLE
Unable to run TCL script in ICICLE Kit

### DISCUSSION THREAD

**Discussion by Usutatsu** [polarfire-soc/polarfire-soc-documentation#242] [How to] [answered]
I am using the icicle kit for development.
I wanted to build a reference design for the 2022.10 version and ran a TCL script on LiberoSoC2022.2.
As a result, I get an error message "ERROR: The command 'download core' failed" and the script is aborted.
What can I do to resolve this?

**Comment by gsriker77** [comment]
(a) Are you running .tcl through GUI (Execute script) (or) command line ?
(b) Make sure you have internet connected
(c) On failing particular IP, You can step as follows 
Please update your IP core (that is failed to download) to latest version at http://www.actel-ip.com/cwps/download_index/DirectCore
Click on Catalog tab->Click settings drop down ->Add core to vault -> Browse this cpz file to configure.

Hope it helps,

**Reply by Usutatsu** [reply]
Thank you for comments.

(a) I'm running .tcl through GUI with command "ctrl+U"
(b) I'm setting proxy and I comfiermed PC can access some internet site.

**Comment by gsriker77** [comment, marked as answer]
Hi,
Looks like below links are referred by .tcl is broken. We will check in background on it.
For time being, you can have remedy as follows :
Remedy : 
Option 1:  Replace links of microchip-ip of both DirectCore/SGCore with below 
http://www.actel-ip.com/cwps/download_index/SgCore
http://www.actel-ip.com/cwps/download_index/DirectCore
Run .tcl again freshly.

Option 2 : 
> Download respective IP (which referred in .tcl, example below) from below link 
Ex: download_core -vlnv {Microsemi:SgCore:PFSOC_INIT_MONITOR:1.0.205} -location {www.microchip-ip.com/repositories/SgCore}
Here it is downloading PFSOC_INIT_MONITOR:1.0.205 which will get from below links
http://www.actel-ip.com/cwps/download_index/SgCore
http://www.actel-ip.com/cwps/download_index/DirectCore
> 'Add core to Valut' , IP that is downloaded
> Comment line in .tcl the IP that is downloaded
> Run .tcl script freshly

Hope it helps,
Regards,
